courseworks/week10/xuanzhaogao/hw10.ipynb

```diff
@@ -0,0 +1,336 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "using LinearAlgebra"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# task 1\n",
+    "\n",
+    "Given $A\\in \\mathbb{R}^{n\\times n}$ and $x, b\\in \\mathbb{R}^n$ Please derive the backward rule of $\\mathcal{L} = \\|Ax - b\\|_2$ either using the chain rules or the perturbative approach (from the last lecture)."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "$$\n",
+    "\\begin{split}\n",
+    "    \\frac{d \\mathcal{L}}{d x} \n",
+    "    & = \\frac{d \\mathcal{L}}{d ||Ax - b||_2} \\frac{d \\sqrt{(Ax - b)^T (Ax - b)}}{d (Ax - b)} \\frac{d (Ax - b)}{dx} \\\\\n",
+    "    & = 1 \\times \\frac{1}{2 ||Ax - b||_2} \\times 2(Ax - b)^T \\times A\\\\\n",
+    "    & = \\frac{(Ax - b)^T \\times A}{||Ax - b||_2}\n",
+    "\\end{split}\n",
+    "$$"
```

Comment:
1. please use $\partial$ for partial derivative.
2. the shape of the output vector is not correct.
```julia
julia> randn(3)' * randn(3, 3)
1×3 adjoint(::Vector{Float64}) with eltype Float64:
 -1.85937  0.156183  -1.7463
```

The result should be transposed.

Reply:
1. The notation for the partial derivative have been revised.
2. I think the output vector given as in this way may make more sense, considering the simple relation:
```math
\delta f = \frac{\partial  f}{\partial x} \delta x~,
```
if $\delta f$ is a scalar and  $\delta x$ is a column, then $\frac{\partial  f}{\partial x}$ should be a row.